packages/types/src/lib/interfaces.ts

```diff
@@ -1891,3 +1901,34 @@ export interface SignatureData {
 export type ClaimsList = {
   [key: string]: SignatureData;
 }[];
+
+export interface MintWithAuthParams {
+  /**
+   * auth method to use for minting
+   */
+  authMethod: AuthMethod;
+
+  /**
+   * Permission scopes:
+   * https://developer.litprotocol.com/v3/sdk/wallets/auth-methods/#auth-method-scopes
+   */
+  scopes: string[] | number[];
+
+  /**
+   * only applies to webauthn auth method
+   */
+  pubkey?: string;
+
+  /**
+   * The Auth ID of the given auth method. If it's custom auth, then it could be
+   * anything.
+   */
+  authId?: Uint8Array;
```

Comment:
why do we need this we have an `authMethod` already which should contain the id?

Reply:
Not with custom auth though?

Reply:
well they could define it however they want for example
```js
{
'autMethodType`: 81234,
'authMerthodId': 'abc123'
}
```
then we convert the `authMethodId` to an `uint8array`

Reply:
hmm the default authMethod interface only has `authMethodType` and `accessToken` no? then the `authId` should be something like `app-id-xxx:user-id-yyy` for the custom auth.

Reply:
Custom Auth Access token can be anything and it's not required that they store it on our chain

Reply:
I think the problem is the only place we have `authMethodId`  as a property on an `authMethod` is in our smart contract implementations for registering it to a pkp. Within the sdk we derive the id we put on chain from the `authMethodId` as an example see the [discord](https://github.com/LIT-Protocol/js-sdk/blob/6af6a7c1847f2881764f99ed07e6777e5bf30372/packages/lit-auth-client/src/lib/providers/DiscordProvider.ts#L98) auth method. 
So we drive the `authMethodId` from the access token if it's an auth method we support. so I don't know if we need this outside of the new `mintWithCustomAuthParams`